File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/VendorInheritNekoDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of vendor_inherit_neko. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class VendorInheritNekoDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final VendorInheritNekoDbm _instance = new VendorInheritNekoDbm();
    private VendorInheritNekoDbm() {}
    public static VendorInheritNekoDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((VendorInheritNeko)et).getInuId(), (et, vl) -> ((VendorInheritNeko)et).setInuId(cti(vl)), "inuId");
        setupEpg(_epgMap, et -> ((VendorInheritNeko)et).getInuName(), (et, vl) -> ((VendorInheritNeko)et).setInuName((String)vl), "inuName");
        setupEpg(_epgMap, et -> ((VendorInheritNeko)et).getInuDate(), (et, vl) -> ((VendorInheritNeko)et).setInuDate(ctld(vl)), "inuDate");
        setupEpg(_epgMap, et -> ((VendorInheritNeko)et).getNekoId(), (et, vl) -> ((VendorInheritNeko)et).setNekoId(cti(vl)), "nekoId");
        setupEpg(_epgMap, et -> ((VendorInheritNeko)et).getNekoName(), (et, vl) -> ((VendorInheritNeko)et).setNekoName((String)vl), "nekoName");
        setupEpg(_epgMap, et -> ((VendorInheritNeko)et).getNekoDate(), (et, vl) -> ((VendorInheritNeko)et).setNekoDate(ctld(vl)), "nekoDate");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "vendor_inherit_neko";
    protected final String _tableDispName = "vendor_inherit_neko";
    protected final String _tablePropertyName = "vendorInheritNeko";
    protected final TableSqlName _tableSqlName = new TableSqlName("vendor_inherit_neko", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnInuId = cci("inu_id", "inu_id", null, null, Integer.class, "inuId", null, false, false, true, "int4", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnInuName = cci("inu_name", "inu_name", null, null, String.class, "inuName", null, false, false, true, "varchar", 2147483647, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnInuDate = cci("inu_date", "inu_date", null, null, java.time.LocalDate.class, "inuDate", null, false, false, false, "date", 13, 0, null, false, null, null, null, null, null, false);
//"df.NullDate"
    protected final ColumnInfo _columnNekoId = cci("neko_id", "neko_id", null, null, Integer.class, "nekoId", null, true, false, true, "int4", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnNekoName = cci("neko_name", "neko_name", null, null, String.class, "nekoName", null, false, false, true, "varchar", 2147483647, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnNekoDate = cci("neko_date", "neko_date", null, null, java.time.LocalDate.class, "nekoDate", null, false, false, false, "date", 13, 0, null, false, null, null, null, null, null, false);
//"df.NullDate"

    /**
     * inu_id: {NotNull, int4(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnInuId() { return _columnInuId; }
    /**
     * inu_name: {NotNull, varchar(2147483647)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnInuName() { return _columnInuName; }
    /**
     * inu_date: {date(13)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnInuDate() { return _columnInuDate; }
    /**
     * neko_id: {PK, NotNull, int4(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNekoId() { return _columnNekoId; }
    /**
     * neko_name: {NotNull, varchar(2147483647)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNekoName() { return _columnNekoName; }
    /**
     * neko_date: {date(13)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNekoDate() { return _columnNekoDate; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnInuId());
        ls.add(columnInuName());
        ls.add(columnInuDate());
        ls.add(columnNekoId());
        ls.add(columnNekoName());
        ls.add(columnNekoDate());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnNekoId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.VendorInheritNeko"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.VendorInheritNekoCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.VendorInheritNekoBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<VendorInheritNeko> getEntityType() { return VendorInheritNeko.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public VendorInheritNeko newEntity() { return new VendorInheritNeko(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((VendorInheritNeko)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((VendorInheritNeko)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}

File: src/dbflute/adf/meta/vendorInheritNekodbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type VendorInheritNekoDbm_T struct {
	df.BaseDBMeta
	ColumnInuId *df.ColumnInfo
	ColumnInuName *df.ColumnInfo
	ColumnInuDate *df.ColumnInfo
	ColumnNekoId *df.ColumnInfo
	ColumnNekoName *df.ColumnInfo
	ColumnNekoDate *df.ColumnInfo
}

func (b *VendorInheritNekoDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *VendorInheritNekoDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var VendorInheritNekoDbm *VendorInheritNekoDbm_T

func Create_VendorInheritNekoDbm() {
	VendorInheritNekoDbm = new(VendorInheritNekoDbm_T)
	VendorInheritNekoDbm.TableDbName = "vendor_inherit_neko"
	VendorInheritNekoDbm.TableDispName = "vendor_inherit_neko"
	VendorInheritNekoDbm.TablePropertyName = "vendorInheritNeko"
	VendorInheritNekoDbm.TableSqlName = new(df.TableSqlName)
	VendorInheritNekoDbm.TableSqlName.TableSqlName = "vendor_inherit_neko"
	VendorInheritNekoDbm.TableSqlName.CorrespondingDbName = VendorInheritNekoDbm.TableDbName

	var vendorInheritNeko df.DBMeta
	vendorInheritNeko = VendorInheritNekoDbm
	VendorInheritNekoDbm.DBMeta=&vendorInheritNeko
	inuIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo inu_id
	inuIdSqlName.ColumnSqlName = "inu_id"
	inuIdSqlName.IrregularChar = false
	VendorInheritNekoDbm.ColumnInuId = df.CCI(&vendorInheritNeko, "inu_id", inuIdSqlName, "", "", "Integer.class", "inuId", "", false, false,true, "int4", 10, 0, "",false,"","", "","","",false,"int64")
	inuNameSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo inu_name
	inuNameSqlName.ColumnSqlName = "inu_name"
	inuNameSqlName.IrregularChar = false
	VendorInheritNekoDbm.ColumnInuName = df.CCI(&vendorInheritNeko, "inu_name", inuNameSqlName, "", "", "String.class", "inuName", "", false, false,true, "varchar", 2147483647, 0, "",false,"","", "","","",false,"string")
	inuDateSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo inu_date
	inuDateSqlName.ColumnSqlName = "inu_date"
	inuDateSqlName.IrregularChar = false
	VendorInheritNekoDbm.ColumnInuDate = df.CCI(&vendorInheritNeko, "inu_date", inuDateSqlName, "", "", "java.time.LocalDate.class", "inuDate", "", false, false,false, "date", 13, 0, "",false,"","", "","","",false,"df.NullDate")
	nekoIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo neko_id
	nekoIdSqlName.ColumnSqlName = "neko_id"
	nekoIdSqlName.IrregularChar = false
	VendorInheritNekoDbm.ColumnNekoId = df.CCI(&vendorInheritNeko, "neko_id", nekoIdSqlName, "", "", "Integer.class", "nekoId", "", true, false,true, "int4", 10, 0, "",false,"","", "","","",false,"int64")
	nekoNameSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo neko_name
	nekoNameSqlName.ColumnSqlName = "neko_name"
	nekoNameSqlName.IrregularChar = false
	VendorInheritNekoDbm.ColumnNekoName = df.CCI(&vendorInheritNeko, "neko_name", nekoNameSqlName, "", "", "String.class", "nekoName", "", false, false,true, "varchar", 2147483647, 0, "",false,"","", "","","",false,"string")
	nekoDateSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo neko_date
	nekoDateSqlName.ColumnSqlName = "neko_date"
	nekoDateSqlName.IrregularChar = false
	VendorInheritNekoDbm.ColumnNekoDate = df.CCI(&vendorInheritNeko, "neko_date", nekoDateSqlName, "", "", "java.time.LocalDate.class", "nekoDate", "", false, false,false, "date", 13, 0, "",false,"","", "","","",false,"df.NullDate")

	VendorInheritNekoDbm.ColumnInfoList = new(df.List)
	VendorInheritNekoDbm.ColumnInfoList.Add(VendorInheritNekoDbm.ColumnInuId)
	VendorInheritNekoDbm.ColumnInfoList.Add(VendorInheritNekoDbm.ColumnInuName)
	VendorInheritNekoDbm.ColumnInfoList.Add(VendorInheritNekoDbm.ColumnInuDate)
	VendorInheritNekoDbm.ColumnInfoList.Add(VendorInheritNekoDbm.ColumnNekoId)
	VendorInheritNekoDbm.ColumnInfoList.Add(VendorInheritNekoDbm.ColumnNekoName)
	VendorInheritNekoDbm.ColumnInfoList.Add(VendorInheritNekoDbm.ColumnNekoDate)


	VendorInheritNekoDbm.ColumnInfoMap=make(map[string]int)
	VendorInheritNekoDbm.ColumnInfoMap["inuId"]=0
		VendorInheritNekoDbm.ColumnInfoMap["inuName"]=1
		VendorInheritNekoDbm.ColumnInfoMap["inuDate"]=2
		VendorInheritNekoDbm.ColumnInfoMap["nekoId"]=3
		VendorInheritNekoDbm.ColumnInfoMap["nekoName"]=4
		VendorInheritNekoDbm.ColumnInfoMap["nekoDate"]=5
	    VendorInheritNekoDbm.PrimaryKey = true
    VendorInheritNekoDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &vendorInheritNeko
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(VendorInheritNekoDbm.ColumnNekoId)

	VendorInheritNekoDbm.PrimaryInfo = new(df.PrimaryInfo)
	VendorInheritNekoDbm.PrimaryInfo.UniqueInfo = ui
	
	var vendorInheritNekoMeta df.DBMeta = VendorInheritNekoDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["VendorInheritNeko"] = &vendorInheritNekoMeta
}
